scripts/setup.sh

```diff
@@ -1,33 +1,53 @@
 #!/bin/bash
+set -e
 
 source lepidopter-fh/etc/default/lepidopter
 source conf/lepidopter-image.conf
 
-# Set preferred compression method. currently zip and xz supported
-compression_method=( zip xz )
 image_file="lepidopter-${LEPIDOPTER_BUILD}-${ARCH}.img"
 
 function usage() {
     echo "usage: setup.sh [options]"
     echo "with no options the script installs the dependencies and builds" \
             "lepidopter image"
-    echo "-c, --compress compress lepidopter image"
+    echo "-c compress lepidopter image with xz or zip compression (eg. -c xz)"
+    echo "-t create a torrent file of the image and the digests"
 }
 
-while [ $# -ne 0 ]; do
-    case $1 in
-        -c | --compress)        shift
-                                compression=1
-                                ;;
-        -h | --help )           usage
-                                exit
-                                ;;
-        * )                     usage
-                                exit 1
+while getopts "c:ht" opt; do
+    case $opt in
+      c)
+        compression_method+=("$OPTARG")
+        ;;
+      t)
+        build_torrent=true
+        ;;
+      h)
+        usage
+        exit 1
```

Comment:
Super bike shedding, but: I'd `exit 0` in this case: if you request usage, and I print usage, in my view that's something that succeeds rather than failing.